js/ui/grid_core/ui.grid_core.utils.js

```diff
@@ -544,5 +547,53 @@ export default {
         }
 
         return 15000000 / this.getPixelRatio(getWindow());
-    }
+    },
+
+    normalizeLookupDataSource(lookup) {
+        let lookupDataSourceOptions;
+        if(lookup.items) {
+            lookupDataSourceOptions = lookup.items;
+        } else {
+            lookupDataSourceOptions = lookup.dataSource;
+            if(isFunction(lookupDataSourceOptions) && !variableWrapper.isWrapped(lookupDataSourceOptions)) {
+                lookupDataSourceOptions = lookupDataSourceOptions({});
+            }
+        }
+
+        return normalizeDataSourceOptions(lookupDataSourceOptions);
+    },
+
+    getWrappedLookupDataSource(column, dataSource, filter) {
+        const lookupDataSourceOptions = this.normalizeLookupDataSource(column.lookup);
+
+        const lookupDataSource = {
+            load: (loadOptions) => {
+                const d = new Deferred();
+                dataSource.load({
+                    ...loadOptions,
+                    filter: this.combineFilters([filter, loadOptions.filter], 'and'),
+                    group: column.dataField || column.name
```

Comment:
Use isExpanded false parameter to avoid all data items loading: https://js.devexpress.com/Documentation/ApiReference/Data_Layer/CustomStore/LoadOptions/#group.